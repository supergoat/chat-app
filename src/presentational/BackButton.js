// @flow
import React, { Component } from 'react';
import type { NavigationScreenProp } from 'react-navigation';
import {
  Image,
  TouchableOpacity
} from 'react-native';
import CAStyleSheet from "../common/CAStyleSheet";
import chevronLeft from '../assets/img/chevron_left.png';

/** ============================================================================
<BackButton
  navigation={object}
/>
--------------------------------------------------------------------------------
Button pressed to navigate viewer to the previous screen
@param { object } navigation - Navigation object to navigate the viewer
============================================================================= */
type Props = {
  navigation: NavigationScreenProp
};
class BackButton extends Component<Props> {
  static __cards__: Function

  render() {
    const { navigation } = this.props;
    return (
      <TouchableOpacity
        onPress={() => navigation.pop()}
        style={styles.back}
      >
        <Image source={chevronLeft} style={styles.backIcon} />
      </TouchableOpacity>
    )
  }
}

/* StyleSheet
============================================================================= */
const styles = CAStyleSheet.create({
  back: {
    paddingRight: 15
  },
  backIcon: {
    height: 20,
    width: 20,
  }
});

/* Playground Cards - the different states of BackButton
============================================================================= */
// BackButton.__cards__ = define => {
//   define("Simple", () => {
//     return <BackButton />
//   });
// };

/* Export
============================================================================= */
export default BackButton;
