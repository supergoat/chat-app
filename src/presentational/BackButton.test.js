// @flow
import React from 'react';
import BackButton from './BackButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('BackButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BackButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays chevronLeft when Platform is iOS', () => {
    const component = shallow(<BackButton />);
    const image = component.find('Image')
    const source = image.prop('source');
    expect(source.testUri.includes('chevron_left.png')).toBe(true)
  });

  it('displays arrowBack when Platform is Android', () => {
    // Set platform to android
    jest.mock('Platform', () => ({
      OS: 'android',
    }));

    const component = shallow(<BackButton />);
    const image = component.find('Image')
    const source = image.prop('source');
    expect(source.testUri.includes('arrow_back.png')).toBe(true)
  });

  it('calls navigation pop from props when BackButton is pressed', () => {
    const navigation = {
      pop: jest.fn(() => console.log('Pop Called'))
    };
    const component = shallow(<BackButton navigation={navigation} />);
    component.find('TouchableOpacity').prop('onPress')()
    expect(navigation.pop.mock.calls.length).toBe(1);
  });
});
