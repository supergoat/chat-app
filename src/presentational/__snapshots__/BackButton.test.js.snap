// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "paddingRight": 15,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/img/chevron_left.png",
      }
    }
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</TouchableOpacity>
`;
